client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
+		expect( container.firstChild ).toHaveAttribute( 'href', '/' );
 	} );
 
-	test( 'should render Card with no href and CTA button with href if href prop is passed and callToAction is also passed', () => {
-		const comp = shallow( <Banner { ...props } href={ '/' } callToAction="Go WordPress!" /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBeNull();
-		expect( comp.find( Card ).props().onClick ).toBeNull();
+	test( 'should render Card with no href and CTA button with href if href prop is passed and callToAction is also passed', async () => {
+		const handleClick = jest.fn();
+		const { container } = render(
+			<Banner { ...props } href={ '/' } callToAction="Go WordPress!" onClick={ handleClick } />
+		);
+
+		await userEvent.click( container.firstChild );
+		await userEvent.click( screen.getByText( 'Go WordPress!' ) );
+
+		expect( handleClick ).toHaveBeenCalledTimes( 1 );
+
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild.href ).toBeUndefined();
 
-		expect( comp.find( Button ) ).toHaveLength( 1 );
-		expect( comp.find( Button ).props().href ).toBe( '/' );
-		expect( comp.find( Button ).props().children ).toBe( 'Go\xA0WordPress!' ); //preventwidows adds \xA0 non-breaking space;
-		expect( comp.find( Button ).props().onClick ).toBe( comp.instance().handleClick );
+		expect( screen.getByText( 'Go WordPress!' ) ).toBeVisible();
+		expect( screen.getByText( 'Go WordPress!' ) ).toHaveAttribute( 'href', '/' );
 	} );
 
-	test( 'should render Card with href and CTA button with no href if href prop is passed and callToAction is also passed and forceHref is true', () => {
-		const comp = shallow(
-			<Banner { ...props } href={ '/' } callToAction="Go WordPress!" forceHref={ true } />
+	test( 'should render Card with href and CTA button with no href if href prop is passed and callToAction is also passed and forceHref is true', async () => {
+		const handleClick = jest.fn();
+		const { container } = render(
+			<Banner
+				{ ...props }
+				href={ '/' }
+				callToAction="Go WordPress!"
+				forceHref={ true }
+				onClick={ handleClick }
+			/>
 		);
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBe( '/' );
-		expect( comp.find( Card ).props().onClick ).toBe( comp.instance().handleClick );
 
-		expect( comp.find( Button ) ).toHaveLength( 1 );
-		expect( comp.find( Button ).props().href ).toBeUndefined();
-		expect( comp.find( Button ).props().children ).toBe( 'Go\xA0WordPress!' ); //preventWidows adds \xA0 non-breaking space;
+		await userEvent.click( container.firstChild );
+
+		expect( handleClick ).toHaveBeenCalledTimes( 1 );
+
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).toHaveAttribute( 'href', '/' );
+
+		expect( screen.getByRole( 'button' ) ).toBeVisible();
+		expect( screen.getByRole( 'button' ).href ).toBeUndefined();
```

Comment:
This will be undefined, because there's no way to access props like this. We should look for the representation of this prop in the UI and test against that instead.